library/ziti_ctrl.c

```diff
@@ -366,6 +374,16 @@ void ziti_ctrl_login(
 }
 
 void ziti_ctrl_current_identity(ziti_controller *ctrl, void(*cb)(ziti_identity_data *, const ziti_error *, void *), void *ctx) {
+    if(ctrl->api_session_token == NULL) {
```

Comment:
refactor into a function, something like `bool verify_api_session(cb, ctx)`

if no session it would call `cb` with error return `false`otherwise return `true`

all these could be changed to 
```if (!verify_session(cb, ctx)) return;```

Reply:
fixed.